resources/tests/EasyCrypt_GenDoc_Tutorial.ec

```diff
@@ -0,0 +1,311 @@
+&*)
+module M = {
+  (*
+    This regular comment appears in the source code
+    for this module in the generated documentation file.
+  *)
+  proc p() : int = {
+    return 1;
+  }
+}.
+
+(*&
+  Theories are special as "documentable" items. They appear as documented items
+  in the generated documentation file for their parent theory __and__
+  receive their own documentation file, in turn documenting all their
+  "documentable" items. This file has a subheading indicating the subtheory and
+  links to entry of the (sub)theory in the parent theory's documentation file.
+
+  The file name for a (sub)theory follows this pattern: `Y!Z`, where `Y`, and
+  `Z` represent the parent theory and (sub)theory, respectively. (This works
+  recursively: `Y` may itself be a (sub)theory of another theory `X`, in which
+  case the file name becomes `X!Y!Z`.)
+
+  The "introductory text" for the (sub)theory, which you would usually put in
+  file documentation comments, is drawn from the regular documentation comments
+  in the parent theory. In the parent theory's documentation file, the
+  (sub)theory's name links to the corresponding (sub)theory documentation file.
+
+  No source code is shown for subtheories in the parent theory's documentation
+  file.
+&*)
+theory T.
+
+(*&
+  This item is documented in the documentation file corresponding to
+  (sub)theory `T'.
+&*)
+type s.
+
+(*&
+  Linking to items is done from the perspective of the outermost theory (`Top`),
+  so names for items within (sub)theories that are not imported must be qualified.
+  In other words, if a (sub)theory is not imported by the outside-most theory,
+  linking requires the item name to be qualified properly, even within the
+  (sub)theory itself. For example, to link to `type s` above, something along
+  the lines of `[go to s in T](>|s)` __does not__ work, but
+  `[go to s in T](>|T.s)` __does__ (proof: [go to s in T](>|T.s)).
+  However, if the (sub)theory would be mported at some point in the
+  outermost theory, `[go to s in T](>|s)` __would work__, provided
+  there are no naming collisions.
+&*)
+op a : s -> s.
+
+(*&
+  Linking to items in a parent theory works as expected. For example `[go to w
+  in parent](>|w)` would create a link to the entry for `w` in the parent
+  theory's documentation file (proof: [go to w in parent](>|w)).
+&*)
+op b : w -> w.
+
+(*&
+  The documentation mechanism for theories works recursively.
+  For example, theory `U` below is treated in the documentation file for `T` in the
+  same way that `T` is treated in the documentation file for the outermost theory.
+  In addition to appearing in `T`'s documentation file, `U` also receives its own
+  documentation file, similar to `T`, but now linking back to `T` rather than the
+  outermost theory.
+&*)
+abstract theory U.
+
+end U.
+
+end T.
+
+
+section.
+
+(*
+  As mentioned before, "scoped" items are never documented, even
+  if their "non-scoped" version are.
+*)
```

Comment:
I think this should be gated by a flag. Obviously, the default case (documenting a library) does not need to reveal any of the proof-only objects that get scoped out. But the secondary use case (creating tutorial material) would be nice to support as well. Perhaps a simple `-tutorial` option flag?